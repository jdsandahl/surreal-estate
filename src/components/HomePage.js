import React from "react";
import "../styles/HomePage.css";

const HomePage = () => {
  return (
    <div className="home-page">
      <aside className="home-page__welcome">
        <h2 className="home-page__heading">Welcome</h2>
        <p>
          From our app you're able to view, search and post properties within
          the UK.
        </p>
        <p>
          Save your favourite properties to view later by logging in with
          Facebook
        </p>
      </aside>
    </div>
  );
};

export default HomePage;
