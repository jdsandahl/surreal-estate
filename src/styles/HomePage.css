.home-page {
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    background-color: var(--secondary-light-color);
    
}

.home-page__heading {
    font-family: "Orbitron", "Open Sans", sans-serif;
}

.home-page__welcome {
    margin: 25px;
    padding: 30px;
    background-color: var(--primary-light-color-trans);
    box-shadow: 0 4px 8px 0 var(--tertiary-dark-color);
    max-width: 500px;
}