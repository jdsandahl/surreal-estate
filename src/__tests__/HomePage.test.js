import React from 'react';
import {render} from "@testing-library/react";
import HomePage from "../components/HomePage";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

describe("HomePage", () => {
    test("the page renders correctly against snapshot", () => {
      const history = createMemoryHistory();
      const { asFragment } = render(
        <Router history={history}>
          <HomePage />
        </Router>
      );
      const component = asFragment();
  
      expect(component).toMatchSnapshot();
    });
  });
