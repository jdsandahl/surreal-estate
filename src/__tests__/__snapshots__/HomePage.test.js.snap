// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage the page renders correctly against snapshot 1`] = `
<DocumentFragment>
  <div
    class="home-page"
  >
    <aside
      class="home-page__welcome"
    >
      <h2
        class="home-page__heading"
      >
        Welcome
      </h2>
      <p>
        From our app you're able to view, search and post properties within the UK.
      </p>
      <p>
        Save your favourite properties to view later by logging in with Facebook
      </p>
    </aside>
  </div>
</DocumentFragment>
`;
